cmd/lifecycle/detector.go

```diff
@@ -88,7 +88,7 @@ func (da detectArgs) detect() (buildpack.Group, platform.BuildPlan, error) {
 		return buildpack.Group{}, platform.BuildPlan{}, err
 	}
 
-	envv := env.NewBuildEnv(os.Environ())
+	envv := env.NewDetectEnv(os.Environ())
```

Comment:
It would be awesome if we could add an acceptance test for this - like have a buildpack that prints its env as part of detect and verify that we don't see `CNB_ASSETS`.

Reply:
(since we already have acceptance tests for the detector)